tests/run.sh

```diff
@@ -104,34 +100,46 @@ function get_requirements() {
     inject_pre_generated_ssh_keys
 }
 
+# Old ways of getting the image, now deprecated, but still needed for images
+# built with thud or older.
+get_ext4_image_deprecated() {
+    if [[ -n "$BUILDDIR" ]]; then
+        cp -f "$BUILDDIR/tmp/deploy/images/$MACHINE_NAME/core-image-full-cmdline-$MACHINE_NAME.ext4" .
+    elif [[ -n "$DOWNLOAD_REQUIREMENTS" ]]; then
+        curl --fail "https://mender.s3-accelerate.amazonaws.com/temp_${MENDER_BRANCH}/core-image-full-cmdline-$MACHINE_NAME.ext4" \
+             -o core-image-full-cmdline-$MACHINE_NAME.ext4 \
+             -z core-image-full-cmdline-$MACHINE_NAME.ext4
+    fi
+}
+
 if [[ $1 == "--get-requirements" ]]; then
     get_requirements
     exit 0
 fi
 
 dd if=/dev/zero of=large_image.dat bs=300M count=0 seek=1
 
-if [[ -n "$BUILDDIR" ]]; then
-    # Get the necessary path directly from the build.
+# mender-stress-test-client is here
+export PATH=$PATH:~/go/bin/
 
-    # On branches without recipe specific sysroots, the next step will fail
-    # because the prepare_recipe_sysroot task doesn't exist. Use that failure
-    # to fall back to the old generic sysroot path.
-    if ( cd "$BUILDDIR" && bitbake -c prepare_recipe_sysroot mender-test-dependencies ); then
-        eval "$(cd "$BUILDDIR" && bitbake -e mender-test-dependencies | grep '^export PATH=')":"$PATH"
-    else
-        eval "$(cd "$BUILDDIR" && bitbake -e core-image-minimal | grep '^export PATH=')":"$PATH"
-    fi
-
-    cp -f "$BUILDDIR/tmp/deploy/images/$MACHINE_NAME/core-image-full-cmdline-$MACHINE_NAME.ext4" .
-
-    # mender-stress-test-client is here
-    export PATH=$PATH:~/go/bin/
-
-elif [[ -n "$DOWNLOAD_REQUIREMENTS" ]]; then
+if [[ -n "$DOWNLOAD_REQUIREMENTS" ]]; then
```

Comment:
I think this is not correct. For a Jenkins build, it will download mender-artifact and mender-stress-client from AWS instead of using the built ones from the current build run.

In most of the cases won't matter, though, but it is a change from the previous logic. Is it intentional?

Reply:
Ah, good catch, I was a bit too quick here. I'm still testing this, so I'll probably get an error soon.